sycl/doc/CompilerAndRuntimeDesign.md

```diff
@@ -549,9 +549,9 @@ is then passed to the offload wrapper tool.
 ##### Checking if the compiler is targeting NVPTX
 
 When the SYCL compiler is in device mode and targeting the NVPTX backend,
-compiler defines the macro `__SYCL_NVPTX__`.
-This macro can safely be used to enable NVPTX specific code path in SYCL
-kernels.
+compiler defines the macro `__NVPTX__` alongside the
```

Comment:
How about - the compiler defines the `__NVPTX__`  and  `__SYCL_DEVICE_ONLY`  macros.

Reply:
Thanks for the suggestion! My initial intent was to illustrate that `__SYCL_DEVICE_ONLY__` is "already known to be used for device compilations", but since there were no preceding mentions of the macro, this no longer made sense. Applied in https://github.com/intel/llvm/pull/3977/commits/f8afe3889a96aef492d027567f5a466a8b9b1d35 + some additional edits.